pkg/server/tls/authority/authority.go

```diff
@@ -107,7 +109,13 @@ func (d *DynamicAuthority) Run(ctx context.Context) error {
 		d.LeafDuration = time.Hour * 24 * 7 // 7d
 	}
 
-	cl, err := kubernetes.NewForConfig(d.RESTConfig)
+	if d.newClient == nil {
+		d.newClient = func(c *rest.Config) (kubernetes.Interface, error) {
+			return kubernetes.NewForConfig(c)
+		}
```

Comment:
```suggestion
		d.newClient = kubernetes.NewForConfig
```

Does this work?

Reply:
Unfortunately, that does not work because https://pkg.go.dev/k8s.io/client-go/kubernetes#NewForConfig returns a *[Clientset](https://pkg.go.dev/k8s.io/client-go/kubernetes#Clientset) instead of the interface. Go does not seem to be smart enough to cast that value without this lambda function.